conans/test/functional/toolchains/google/test_bazel.py

```diff
@@ -119,152 +97,80 @@ def test_transitive_consuming():
-    from conan.tools.build import cross_building
-
-
-    class OpenSSLTestConan(ConanFile):
-        settings = "os", "compiler", "build_type", "arch"
-        generators = "BazelToolchain", "BazelDeps", "VirtualBuildEnv", "VirtualRunEnv"
-        apply_env = False
-
-        def requirements(self):
-            self.requires(self.tested_reference_str)
-
-        def build(self):
-            bazel = Bazel(self)
-            bazel.build(target="//main:example")
-
-        def layout(self):
-            bazel_layout(self)
-
-        def test(self):
-            if not cross_building(self):
-                cmd = os.path.join(self.cpp.build.bindirs[0], "main", "example")
-                self.run(cmd, env="conanrun")
-    """)
-
-    test_bazel_build = textwrap.dedent("""
-        load("@rules_cc//cc:defs.bzl", "cc_binary")
-
-        cc_binary(
-            name = "example",
-            srcs = ["example.cpp"],
-            deps = [
-                "@openssl//:openssl",
-            ],
+        """)
+        bazel_build_linux = textwrap.dedent("""\
+        load("@rules_cc//cc:defs.bzl", "cc_library")
+
+        cc_library(
+            name = "openssl",
+            srcs = ["openssl.cpp"],
+            hdrs = ["openssl.h"],
+            deps = [ "@zlib//:zlib" ]
         )
         """)
+        bazel_build = textwrap.dedent("""\
+        load("@rules_cc//cc:defs.bzl", "cc_library")
+
+        cc_library(
+            name = "openssl",
+            srcs = ["openssl.cpp"],
+            hdrs = ["openssl.h"],
+            deps = [ "@zlib//:zlib" ]
+        )
 
-    client.save({"conanfile.py": conanfile,
-                 "main/BUILD": bazel_build,
-                 "main/openssl.cpp": openssl_c if platform.system() != "Windows" else openssl_c_win,
-                 "main/openssl.h": openssl_h,
-                 "WORKSPACE": bazel_workspace,
-                 "test_package/conanfile.py": test_conanfile,
-                 "test_package/main/example.cpp": test_example,
-                 "test_package/main/BUILD": test_bazel_build,
-                 "test_package/WORKSPACE": bazel_workspace
-                 })
-
-    client.run("create .")
-    assert "Calling OpenSSL function with define MY_VALUE and other define 2" in client.out
-    assert "zlib/1.2.11: Hello World Release!"
+        cc_shared_library(
+            name = "openssl_shared",
+            shared_lib_name = "libopenssl.{}",
+            deps = [":openssl"],
+        )
+        """.format("dylib" if os_ == "Darwin" else "dll"))
+        openssl_c = textwrap.dedent("""
+        #include <iostream>
+        #include "openssl.h"
+        #include "zlib.h"
+        #include <math.h>
+
+        void openssl(){
+            std::cout << "openssl() First define " << MY_DEFINE << " and other define " << MY_OTHER_DEFINE << "\\n";
+            zlib();
+            // This comes from the systemlibs declared in the zlib
+            sqrt(25);
```

Comment:
not okay to ignore return value (compiler may optimize out if there are no side-effects)